lib/utils.js

```diff
@@ -89,8 +84,28 @@ var merge = function merge(target, source, options) {
         return target;
     }
 
-    return Object.keys(source).reduce(function (acc, key) {
-        var value = source[key];
+    var mergeTarget = target;
+    var mergeSource = source;
+
+    if (isArray(target) && !isArray(source)) {
+        var restrictedKeys = new Set(Object.getOwnPropertyNames(Array.prototype));
```

Comment:
this is also a breaking change; `qs` supports node 0.4, which does not have Set.